### PR TITLE
PRC-157 : Build the PROD environment

### PR DESCRIPTION
**Pull request by prabash-moj**
PRC-157 : Build the PROD environment

### CODE REVIEW

File: namespaces/live.cloud-platform.service.justice.gov.uk/hmpps-contacts-prod/resources/serviceaccount-circleci.tf [hunk 1]
```diff
@@ -0,0 +1,9 @@
+module "serviceaccount_circleci" {
+  source                            = "github.com/ministryofjustice/cloud-platform-terraform-serviceaccount?ref=1.1.0"
+  namespace                         = var.namespace
+  kubernetes_cluster                = var.kubernetes_cluster
+  serviceaccount_token_rotated_date = "19-04-2024"
+  serviceaccount_name               = "circleci"
+  role_name                         = "circleci"
+  rolebinding_name                  = "circleci"
```
Comment (timharrison-moj) [inline namespaces/live.cloud-platform.service.justice.gov.uk/hmpps-contacts-prod/resources/serviceaccount-circleci.tf:8]: This file is very different from the preprod one?

**Review by timharrison-moj — COMMENTED** [review]
LGTM - apart from the serviceaccount-circleci.tf file?

File: namespaces/live.cloud-platform.service.justice.gov.uk/hmpps-contacts-prod/resources/serviceaccount-circleci.tf (hunk 1 above)
Comment (prabash-moj) [reply namespaces/live.cloud-platform.service.justice.gov.uk/hmpps-contacts-prod/resources/serviceaccount-circleci.tf:8]: yeah it is copied from the prod folder of the book a video link project .

Comment (prabash-moj) [reply namespaces/live.cloud-platform.service.justice.gov.uk/hmpps-contacts-prod/resources/serviceaccount-circleci.tf:8]: but booth pre-prod and prod files in that project are similar, our preprod file additionally have this   serviceaccount_rules = local.sa_rules and object associated with that. not sure if they are inherited from somewhere or needed anymore?

Comment (prabash-moj) [reply namespaces/live.cloud-platform.service.justice.gov.uk/hmpps-contacts-prod/resources/serviceaccount-circleci.tf:8]: it looks like `serviceaccount_rules` variable is not needed and inherits from the parent

Comment (timharrison-moj) [reply namespaces/live.cloud-platform.service.justice.gov.uk/hmpps-contacts-prod/resources/serviceaccount-circleci.tf:8]: Go for it.

**Review by jaskaransarkaria — APPROVED** [review]

**Pull request merged**